# My First Guided Project with Dataquest

In this project, I will analyze data from to figure out if there is a trend of applications that is more well-received among users.

The goal for this project is to analyze data to help our developers understand what type of apps are likely to attract more user to boost our revenue. 

In [1]:
from csv import reader 

openedfile = open('googleplaystore.csv')
readfile = reader(openedfile)
android = list(readfile)


openedfile = open('AppleStore.csv')
readfile = reader(openedfile)
ios = list(readfile)


In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

explore_data(ios, 0, 5, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 7198
Number of columns: 16


Number of rows and columns for AppleStore - 7197, 16

Number of rows and columns for Google Play Store - 10841, 16

Important columns that could potential aid us in our analysis are 

1. App Name
2. Price
3. Average User Rating Count
4. Language 
5. Age
6. Genre

You can find a clearer documentation of the column names [here](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps)

In the [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion), it is mentioned by users that there is a missing input in line 10473. Let's investigate if this is true.

In [3]:
print(android[0])
print('\n')
print(android[10473])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


It can be observed in the 3rd reading that the `Rating` is 19. However, it is clear that the maximum rating for Google Play Store is 5. Hence, there is an incorrect data inputted.

I shall delete the row with the incorrect data present

In [4]:
del android[10473]

Let's check for any duplicate values present in the data set.

In [5]:
unique_entry = []
duplicate_entry = []
for row in android:
    app = row[0]
    if app in unique_entry:
        duplicate_entry.append(app)
    else:
        unique_entry.append(app)
print('Number of duplicate apps:', len(duplicate_entry))

Number of duplicate apps: 1181


Duplicates present in the data set would not be removed at random. 
I will use the number of reviews as the criteria when removing duplicates. Same application with more number of reviews signifies a later version and hence would be kept within the data set.

To achieve our aim, we will 
- Create a dictionary where each key is a unique app name, and the value is the highest number of reviews of that app.
- We will use the dictionary to create a new data set, which will have only one entry per application - the one with the highest number of reviews. 

In [6]:
reviews_max = {}

for row in android[1:]:
    name = row[0]
    n_reviews = float(row[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

print('Expected Length:', len(android[1:]) - 1181)
print('Actual Length: ',len(reviews_max))

Expected Length: 9659
Actual Length:  9659


In [7]:
android_clean = []
already_added = []

for row in android[1:]:
    name = row[0]
    n_reviews = float(row[3])
    
    if (n_reviews == reviews_max[name]) and (name not in already_added):
        android_clean.append(row)
        already_added.append(name)
        
explore_data(android_clean, 0,4, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


Number of rows: 9659
Number of columns: 13


Check the number of duplicate applications in the Apple Store

In [8]:
unique_entry1 = []
duplicate_entry1 = []
for row in ios[1:]:
    app = row[0]
    if app in unique_entry:
        duplicate_entry1.append(app)
    else:
        unique_entry1.append(app)
print('Number of duplicate apps:', len(duplicate_entry1))

Number of duplicate apps: 0


No duplicate entries found in the Apple Store

Now, we will proceed to remove non-English applications as we only intend to produce applications for the English audience

In [9]:
def englishcheck(language):
    for letter in language:
        if ord(letter) > 127:
            return False
        
    return True

print(englishcheck('Instagram'))
print(englishcheck('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(englishcheck('Docs To Go™ Free Office Suite'))
print(englishcheck('Instachat 😜'))

True
False
False
False


As seen, `False` is returned even though the name of the application inputted is in English. This is due to the emoji and "tm" sign not being recognised by the system, hence returning us with an incorrect boolean. We will amend the function created to prevent this problem from arising.

In [10]:
def englishcheck(language):
    ascii = 0
    for letter in language:
        if ord(letter) > 127:
            ascii += 1
            if ascii > 3:
                return False
    return True

print(englishcheck('Instagram'))
print(englishcheck('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(englishcheck('Docs To Go™ Free Office Suite'))
print(englishcheck('Instachat 😜'))

True
False
True
True


Using this newly edited function, I will filter out the rows of English Applications

In [11]:
android_english = []

for app in android_clean:
    name = app[0]
    if englishcheck(name):
        android_english.append(app)
    
explore_data(android_english, 0,5, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Number of rows: 9614
Number of columns: 13


In [12]:
ios_english = []
for row in ios[1:]:
    name = row[1]
    if englishcheck(name):
        ios_english.append(row)

In [13]:
explore_data(ios_english, 0,3,True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 6183
Number of columns: 16


We can see that we're left with 9614 Android apps and 6183 iOS apps.

Since we are only interested in building free apps, it is not of our concern to analyze paid apps. Therefore, it is of best interest to isolate the free apps in the data set. 

In [14]:
print(android[0])
print(ios[0])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [15]:
print(android_english[:3])

[['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up'], ['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up'], ['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']]


IOS price index - 4

Android price index - 7 


In [16]:
android_free = [] 

for row in android_english:
    price = row[7]
    
    if price == '0':
        android_free.append(row)
        
explore_data(android_free, 0, 3 , True)


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 8864
Number of columns: 13


In [17]:
ios_free  = []

for row in ios_english:
    price = row[4]
    
    if price == '0.0':
        ios_free.append(row)
        
explore_data(ios_free, 0, 3 , True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 3222
Number of columns: 16


Now, we can see that we are left with 3222 IOS apps and 8864 Android apps for our analysis

To ensure our app is profitable, we will make use of a validation strategy. 

Our final goal is make an app which would be successful on both markets - IOS and Android. Therefore, we are firstly going to launch a minimal verison of the app on the Google Play Store, if the app is well-received among the users, we then develop it further. If we see profits after a span of 6 months, we will then build an IOS version of the app and add it on the App Store. 

Thus, we will now analyze apps that are profitable on both the Android and IOS application stores. We will get started by highlighting the genres more comonly found in each market. 


In [18]:
print(ios[0])
print(android[0])


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


The index of the genre for both platforms are

IOS - 11 `prime_genres`

Android 
-4 `Genre`
 1 `Category`

Now, we are going to build a table to view the percentages of genres present in the respective application stores. We will of a frequency table to carry this process out. Afterwards, we will make use of a list of tuples to sort this frequency table to view the percentages in a descending order. 

In [19]:
def freq_table(dataset, index):
    table ={}
    for row in dataset:
        name = row[index]
        if name in table:
            table[name] += 1
        else: 
            table[name] = 1 
    return table

    

In [20]:
def display_table(dataset,index):
    table = freq_table(dataset,index)
    table_display = [] 
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [21]:
display_table(ios_free, 11)

Games : 1874
Entertainment : 254
Photo & Video : 160
Education : 118
Social Networking : 106
Shopping : 84
Utilities : 81
Sports : 69
Music : 66
Health & Fitness : 65
Productivity : 56
Lifestyle : 51
News : 43
Travel : 40
Finance : 36
Weather : 28
Food & Drink : 26
Reference : 18
Business : 17
Book : 14
Navigation : 6
Medical : 6
Catalogs : 4


In [22]:
display_table(android_free, -4)

Tools : 749
Entertainment : 538
Education : 474
Business : 407
Productivity : 345
Lifestyle : 345
Finance : 328
Medical : 313
Sports : 307
Personalization : 294
Communication : 287
Action : 275
Health & Fitness : 273
Photography : 261
News & Magazines : 248
Social : 236
Travel & Local : 206
Shopping : 199
Books & Reference : 190
Simulation : 181
Dating : 165
Arcade : 164
Video Players & Editors : 157
Casual : 156
Maps & Navigation : 124
Food & Drink : 110
Puzzle : 100
Racing : 88
Role Playing : 83
Libraries & Demo : 83
Auto & Vehicles : 82
Strategy : 81
House & Home : 73
Weather : 71
Events : 63
Adventure : 60
Comics : 54
Beauty : 53
Art & Design : 53
Parenting : 44
Card : 40
Casino : 38
Trivia : 37
Educational;Education : 35
Board : 34
Educational : 33
Education;Education : 30
Word : 23
Casual;Pretend Play : 21
Music : 18
Racing;Action & Adventure : 15
Puzzle;Brain Games : 15
Entertainment;Music & Video : 15
Casual;Brain Games : 12
Casual;Action & Adventure : 12
Arcade;Action & Advent

In [23]:
display_table(android_free, 1)

FAMILY : 1676
GAME : 862
TOOLS : 750
BUSINESS : 407
LIFESTYLE : 346
PRODUCTIVITY : 345
FINANCE : 328
MEDICAL : 313
SPORTS : 301
PERSONALIZATION : 294
COMMUNICATION : 287
HEALTH_AND_FITNESS : 273
PHOTOGRAPHY : 261
NEWS_AND_MAGAZINES : 248
SOCIAL : 236
TRAVEL_AND_LOCAL : 207
SHOPPING : 199
BOOKS_AND_REFERENCE : 190
DATING : 165
VIDEO_PLAYERS : 159
MAPS_AND_NAVIGATION : 124
FOOD_AND_DRINK : 110
EDUCATION : 103
ENTERTAINMENT : 85
LIBRARIES_AND_DEMO : 83
AUTO_AND_VEHICLES : 82
HOUSE_AND_HOME : 73
WEATHER : 71
EVENTS : 63
PARENTING : 58
ART_AND_DESIGN : 57
COMICS : 55
BEAUTY : 53


**IOS APPLICATIONS**

Looking at the most common genres present in the IOS appstore, Games top the charts and Entertainment comes next. 

The number of game applications found in the IOS app store is significantly more than the rest of the other genres, having 1874 applications that fall under its category, while the category second to it, entertainment, only has 254 such applications. 

Most of the applications found in the IOS app store seems to be for entertainment purposes - the top 3 apps being for entertainment purposes with Games being more significantly produced then the other applications.

However, this frequency table may be inaccurate to a certain degree.
This is because the number of applications falling under a certain genre does not neccessarily signify that the demand of that application is higher than the rest. This could be due to a major application, for instance Netflix, occupying huge market share amongst the Entertainment applicatins, causing it to be difficult for new applications to enter the market, thus seeing less developers producing entertainment applications. Whereas, users may always be looking for new games on their devices, hence having the tendency to download more games. This increases the demand for games which cause more developers to produce more of it. 


**ANDROID APPLICATIONS**

Viewing the android application genres, there is a better spread amongst the different genres as compared to the IOS applications.  

The most common category in the android play store is family, with games in second place. 
The most common genre in the android play store is Tools, with entertainment tailing close behind in second place. 

Android play store seems to have more diversified options as each genre seems to possess many apps. This differs from the IOS store where applications other than the top few genres possess very few options. 

However, I am still unable to recommend an app profile based on the data I have gotten. The data that has been produce does not accurately show the popularity of the applications - more applications does not necessarily signify a higher popularity as mentioned earlier. Therefore, it is pertinent that we get data of which genre has the most users to accurately potray the market demand for each genre.

We will move on to calculate the number of users for each genre of application

# Most popular IOS apps by Genre

In [25]:
prime_genre_table = freq_table(ios_free,11)
print(prime_genre_table)

{'Social Networking': 106, 'Photo & Video': 160, 'Games': 1874, 'Music': 66, 'Reference': 18, 'Health & Fitness': 65, 'Weather': 28, 'Utilities': 81, 'Travel': 40, 'Shopping': 84, 'News': 43, 'Navigation': 6, 'Lifestyle': 51, 'Entertainment': 254, 'Food & Drink': 26, 'Sports': 69, 'Book': 14, 'Finance': 36, 'Education': 118, 'Productivity': 56, 'Business': 17, 'Catalogs': 4, 'Medical': 6}


In [30]:
for genre in prime_genre_table:
    total = 0
    len_genre = 0 
    
    for row in ios_free:
        genre_app = row[11]
        if genre_app == genre:
            ratings = float(row[5])
            total += ratings
            len_genre += 1
            
    average_user_rating = total/len_genre
    print(genre, ':', average_user_rating)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


In [56]:
for row in ios_free:
    if row[11] == 'Book':
        print (row[1],':', row[5])

Kindle – Read eBooks, Magazines & Textbooks : 252076
Audible – audio books, original series & podcasts : 105274
Color Therapy Adult Coloring Book for Adults : 84062
OverDrive – Library eBooks and Audiobooks : 65450
HOOKED - Chat Stories : 47829
BookShout: Read eBooks & Track Your Reading Goals : 879
Dr. Seuss Treasury — 50 best kids books : 451
Green Riding Hood : 392
Weirdwood Manor : 197
MangaZERO - comic reader : 9
ikouhoushi : 0
MangaTiara - love comic reader : 0
謎解き : 0
謎解き2016 : 0


I would recommend Book as the app profile of our new application.

The genre possesses nearly 40 thousand number of users on average which is substantial in relative to the other genres. There are few number of applications present under this genre, meaning we face fewer competition in this market, also there are few "giants" in this genre as compared to Social Networking where there are established platforms such as Facebook which makes it harder to compete with. As the main source of revenue for the application is in app advertisements, it is much easier to have advertisements present in book advertisements without causing users to be frustrated. 

# Most Popular Android Apps by Genre

We analyze the average number of users based on the different categories in the Play Store. We will edit our the install numbers and use the estimated values of installation to get the average results. This will cause our data gotten to be inaccurate to a small extent.

In [45]:
category_freqtable = freq_table(android_free, 1)
print(category_freqtable)

{'ART_AND_DESIGN': 57, 'AUTO_AND_VEHICLES': 82, 'BEAUTY': 53, 'BOOKS_AND_REFERENCE': 190, 'BUSINESS': 407, 'COMICS': 55, 'COMMUNICATION': 287, 'DATING': 165, 'EDUCATION': 103, 'ENTERTAINMENT': 85, 'EVENTS': 63, 'FINANCE': 328, 'FOOD_AND_DRINK': 110, 'HEALTH_AND_FITNESS': 273, 'HOUSE_AND_HOME': 73, 'LIBRARIES_AND_DEMO': 83, 'LIFESTYLE': 346, 'GAME': 862, 'FAMILY': 1676, 'MEDICAL': 313, 'SOCIAL': 236, 'SHOPPING': 199, 'PHOTOGRAPHY': 261, 'SPORTS': 301, 'TRAVEL_AND_LOCAL': 207, 'TOOLS': 750, 'PERSONALIZATION': 294, 'PRODUCTIVITY': 345, 'PARENTING': 58, 'WEATHER': 71, 'VIDEO_PLAYERS': 159, 'NEWS_AND_MAGAZINES': 248, 'MAPS_AND_NAVIGATION': 124}


In [52]:
for category in category_freqtable:
    total  = 0
    len_category = 0 
    
    for row in android_free:
        category_app = row[1]
        if category_app == category:
            installs = row[5]
            installs = installs.replace('+', '')
            installs = installs.replace(',','')
            total += float(installs)
            len_category += 1
    
    average_user_install = total/len_category
    print(category, ':',average_user_install)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

As our aim is to launch a product that would be profitable in both the IOS app store and the Google Play Store, let's take a look at the market for Books in the Google Play Store.

In [57]:
for row in android_free:
    if row[1] == 'BOOKS_AND_REFERENCE':
        print(row[0], ':', row[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

This significantly more competition felt in the Google Play Store. This might be due to Android combine the Book category together with the Reference Catetgory of the IOS. Let's take a look at the major competitors

In [58]:
for row in android_free:
    if row[1] == 'BOOKS_AND_REFERENCE' and (row[5] == '1,000,000,000+'
                                           or row[5] == '500,000,000+'
                                           or row[5] == '100,000,000+'):
        print(row[0], ':', row[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


Google Play Store has a huge competitor in Google Play Books that is seemingly doing significantly better than the other applications that fall under this category. This can cause it to be very difficult for our app to be profitable in this genre of the market. 